src/main/java/org/myrobotlab/service/Gpt3.java

```diff
@@ -0,0 +1,255 @@
+package org.myrobotlab.service;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.myrobotlab.codec.CodecUtils;
+import org.myrobotlab.framework.Service;
+import org.myrobotlab.framework.interfaces.Attachable;
+import org.myrobotlab.logging.Level;
+import org.myrobotlab.logging.LoggerFactory;
+import org.myrobotlab.logging.LoggingFactory;
+import org.myrobotlab.programab.Response;
+import org.myrobotlab.service.config.Gpt3Config;
+import org.myrobotlab.service.data.Utterance;
+import org.myrobotlab.service.interfaces.ResponsePublisher;
+import org.myrobotlab.service.interfaces.TextListener;
+import org.myrobotlab.service.interfaces.TextPublisher;
+import org.myrobotlab.service.interfaces.UtteranceListener;
+import org.myrobotlab.service.interfaces.UtterancePublisher;
+import org.slf4j.Logger;
+
+/**
+ * https://beta.openai.com/account/api-keys
+ * 
+ * https://beta.openai.com/playground
+ * 
+ * https://beta.openai.com/examples
+ * 
+ * <pre>
+     * curl https://api.openai.com/v1/completions \
+      -H "Content-Type: application/json" \
+      -H "Authorization: Bearer $OPENAI_API_KEY" \
+      -d '{
+      "model": "text-ada-001",
+      "prompt": "why is the sun so bright ?\n\nThe sun is bright because it is a star. It is born on theentlement, or radiance, which is the product of the surface area of the sun's clouds and the surface area of the sun's atmosphere.",
+      "temperature": 0.7,
+      "max_tokens": 256,
+      "top_p": 1,
+      "frequency_penalty": 0,
+      "presence_penalty": 0
+    }'
+ * 
+ * </pre>
+ * 
+ * @author GroG
+ *
+ */
+public class Gpt3 extends Service implements TextListener, TextPublisher, UtterancePublisher, UtteranceListener, ResponsePublisher {
+
+  private static final long serialVersionUID = 1L;
+
+  public final static Logger log = LoggerFactory.getLogger(Gpt3.class);
+
+  protected String currentChannel;
+
+  private String currentBotName;
+
+  private String currentChannelName;
+
+  private String currentChannelType;
+
+  public Gpt3(String n, String id) {
+    super(n, id);
+  }
+
+  public Response getResponse(String text) {
+
+    try {
+
+      Gpt3Config c = (Gpt3Config) config;

```

Comment:
I don't like that this needs to be type casted here on each response.

Reply:
The framework sets the value ... all the services don't need to worry about handling the config
the getConfig/apply is handled completely by the framework.
To do so its a single field in the Service.java definition is needed.   If not this,
what would you suggest ?